### PR TITLE
update Up and Running with oVirt and Gluster to 4.1.1

### PR DESCRIPTION
**Pull request by jasonbrooks**
I confirm that this pull request was submitted according to the contribution guidelines: @jasonbrooks

This pull request needs review by: (please @mention the reviewer if relevant)

This replaces PR #804

### CODE REVIEW

**Comment by jasonbrooks** [comment]
@mykaul

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+Read on to learn about my favored way of running oVirt.
+
+READMORE
+
+## oVirt, Glusterized
+
+### Prerequisites
+
+__Hardware:__ You’ll need three machines with plenty of RAM and processors with [hardware virtualization extensions](http://en.wikipedia.org/wiki/X86_virtualization#Hardware-assisted_virtualization). Physical machines are best, but you can test oVirt using [nested KVM](http://community.redhat.com/blog/2013/08/testing-ovirt-3-3-with-nested-kvm/) as well. I've written this howto using VMs running on my "real" oVirt+Gluster install.
```
Comment (mykaul) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: I run 3 VMs with hosted-engine on my 8GB laptop. I'd remove 'plenty', with something more specific. Say, 16GB or more.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+### Installing oVirt with hosted engine
+
+I'm starting out with three test machines with 16 GB of RAM and 4 processor cores, running oVirt Node 4.1.1. I actually do the testing for this howto in VMs hosted on my "real" oVirt setup, but that "real" setup resembles what I describe below.
+
+I've identified a quartet of static IP address on my network to use for this test (three for my virt hosts, and one for the hosted engine). I've set up the DNS server in my lab to make these IPs resolve properly, but you can also edit the /etc/hosts files on your test machines for this purpose.
+
+Our host1 will need to be able to access itself and the other two nodes via passwordless ssh:
+
+```
+[host1]# ssh-keygen
```
Comment (mykaul) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: perhaps worth writing the full command for 'ssh-keygen' ?

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+[host1]# ssh-copy-id root@$HOST2
+
+[host1]# ssh-copy-id root@$HOST3
+```
+
+Next, open up a web browser and visit your first host at port 9090 to access the cockpit web interface. Log in with the machine's root account, click the "Virtualization" tab at the top of the screen, and then click the "Hosted Engine" link in the left sidebar. Select the radio button next to "Hosted Engine with Gluster" and hit the "Start" button.
+
+![](uarwo41-cockpit-1.png)
+
+The dialog window that appears contains a series of steps through which you provide gdeploy with the information it needs to configure your three nodes for running ovirt with gluster storage, starting with the hosts you want to configure.
```
Comment (mykaul) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: I think this is the 1st time you mention gdeploy - perhaps worth explaining what is it?

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+Click next to accept the defaults in step two, and then in step three, specify the gluster volumes you want to create. The cockpit gdeploy plugin autofills some values here, including a volume for the engine, a data volume, and a second data volume called vmstore. The storage domains you'll need for a minimal ovirt install are engine, data, export and iso, and so these are the ones I create: 
+
+![](uarwo41-gdeploy-3.png)
+
+Click "Next" to hit step four, where we'll specify the brick locations for our volumes. Again, the plugin prefills some values here, which aren't likely to be correct for your environment, so pay close attention here. In my test environment, I'm using one additional disk for my gluster data, `/dev/sdb`, and I'm specifying one brick per host per volume: 
+
+![](uarwo41-gdeploy-4.png)
+
+In the final, "Review" step, I found it necessary to click "Edit" and the following operation after the `script1` step:
```
Comment (mykaul) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Is there a bug for this? I'm not aware of this requirement?

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+After making this edit, hit the "Save" button, and then hit "Deploy" to kick off the deployment process. This process will take some time to complete, as gdeploy installs required packages and configures gluster volumes and their underlying storage.
+
+![](uarwo41-gdeploy-6.png)
+
+### Hosted engine setup
+
+Now, click the "Continue to Hosted Engine Deployment" button to begin configuring your hosted engine. After accepting the default "Yes" and clicking the "Next" button to begin the process, the install will offer to download the oVirt engine appliance image. Click "Next" to proceed.
+
+The installer will ask if you want to configure your host and cluster for Gluster. Again, click "Next" to proceed. In some of my tests, the installer failed at this point, with an error message of `Failed to execute stage 'Environment customization'`. When I encountered this, I clicked "Restart Setup", repeated the above steps, and was able to proceed normally.
```
Comment (mykaul) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Is there a bug for this issue?

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+Now, click the "Continue to Hosted Engine Deployment" button to begin configuring your hosted engine. After accepting the default "Yes" and clicking the "Next" button to begin the process, the install will offer to download the oVirt engine appliance image. Click "Next" to proceed.
+
+The installer will ask if you want to configure your host and cluster for Gluster. Again, click "Next" to proceed. In some of my tests, the installer failed at this point, with an error message of `Failed to execute stage 'Environment customization'`. When I encountered this, I clicked "Restart Setup", repeated the above steps, and was able to proceed normally. 
+
+You'll need to specify the `glusterfs` storage type, and then supply the path to your Gluster volume, which should be something like `host1:engine`.
+
+Next, we need to specify which network interface to use for oVirt's management network, and whether the installer should configure our firewall. In some of my tests with oVirt Node, the management network setup step failed due to the presence of an ifcfg-eth0.bak file. When I encountered this issue, I removed the file from each of my hosts, restarted the process, and was able to proceed.
```
Comment (mykaul) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Is there a bug for this ifcfg-eth0.bak ?

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+## Configuring hosts two and three for Hosted Engine
+
+Head over to the Hosts tab, select host two, and in the toolbar below the tabs, click Management, and then Maintenance, and hit OK in the dialog box that appears next. Once the host is in maintenance mode, click Installation, and then Reinstall in the toolbar. In the dialog that appears next, click "Hosted Engine," and then, in the drop down box, choose DEPLOY, and hit OK. After that process completes, repeat the process on host three.
+
+![](uarwo41-deploy-hosted.png)
+
+Once all three hosts are back up, you should be able to bring down any one of the hosts at a time without losing access to the management engine or to your VM storage. 
+
+## Running your first VM
```
Comment (mykaul) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Perhaps first show that the Hosted Engine VM is available in the oVirt UI?

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+![](uarwo40-storage-net-b.png)
+
+## Maintenance, failover, and storage
+
+The key thing to keep in mind regarding host maintainence and downtime is that this converged three node system relies on having at least two of the nodes up at all times. If you bring down two machines at once, you'll run afoul of the Gluster quorum rules that guard us from split-brain states in our storage, the volumes served by your remaining host will go read-only, and the VMs stored on those volumes will pause and require a shutdown and restart in order to run again.
+
+The oVirt engine pays attention to the state of its configured gluster volumes, and will warn you if certain actions will run afoul of quorum rules or if your volumes have pending healing operations. 
+
+You can bring a single machine down for maintenance by first putting the system into maintenance mode from the oVirt console, and updating, rebooting, shutting down, etc. as desired.
```
Comment (mykaul) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Worth expanding how to do it?

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
+---
+title: Up and Running with oVirt 4.1 and Gluster Storage
+author: jbrooks
+date: 2017-03-29 08:00:00 UTC
+tags: centos, gluster, ovirt, hyperconverged
+comments: true
+published: true
+---
+
+Last month, the oVirt Project [shipped version 4.1](https://www.ovirt.org/release/4.1.0/) of its open source virtualization management system. With a new release comes an update to this howto for running oVirt together with Gluster storage using a trio of servers to provide for the system's virtualization and storage needs, in a configuration that allows you to take one of the three hosts down at a time without disrupting your running VMs.
```
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Changed to email announcement.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+**Important Note:** I want to stress that while Red Hat has recently begun to sell and support a converged [virtualization and storage configuration](https://access.redhat.com/articles/2360321) on a limited basis, the converged oVirt/Gluster setup I describe here should be considered somewhat bleeding edge.
```
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: I'd added this to past versions of this howto by request. I'll remove.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+If you're looking instead for a simpler, single-machine option for trying out oVirt, your best bet is the <a href="http://www.ovirt.org/OVirt_Live">oVirt Live ISO</a>. This is a LiveCD image that you can burn onto a blank CD or copy onto a USB stick to boot from and run oVirt. This is probably the fastest way to get up and running, but once you're up, this is definitely a low-performance option, and not suitable for extended use or expansion.
+
+Read on to learn about my favored way of running oVirt.
```
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: These are synonyms, but I'll change.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 1 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Will change.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 3 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Added.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 4 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: The hyperconverged gdeploy example still has it: https://github.com/gluster/gdeploy/blob/master/examples/hc.conf#L67. I don't know why it isn't in here, but my /dev/sdb isn't found w/o disabling multipath. I'll look into filing an issue.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 5 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: I haven't filed one, but I will.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 6 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: I'll look and file if need be.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 7 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: I'm not sure what you mean -- like a screen shot showing the entry for the running Hosted Engine VM?

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 8 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: Added some text on this.

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 4 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: I filed: https://bugzilla.redhat.com/show_bug.cgi?id=1438596

File: source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md (hunk 5 above)
Comment (jasonbrooks) [inline source/blog/2017-03-29-up-and-running-with-ovirt-4-1-and-gluster-storage.html.md]: filed: https://bugzilla.redhat.com/show_bug.cgi?id=1438604

**Comment by jasonbrooks** [comment]
@JohnMarksRH I'd like to get this up on the blog soon, unless there's a 4.1.1.1.1 that we should be waiting for ;)

**Comment by JohnMarksRH** [comment]
@jasonbrooks @mykaul Can you please clarify what is required of me. Thanks

**Comment by bproffitt** [comment]
@JohnMarksRH He wants you to merge the blog so it can post. I fixed the frontmatter date and once the CI checks are done, you or IO can merge.

**Pull request merged**